### PR TITLE
PR: Fix restart mechanism for the Windows standalone installer (Installers)

### PR DESCRIPTION
**Pull request by dalthviz**
## Description of Changes

Be sure to use the script the standalone installer uses in the shortcut to start and don't remove installer `pkgs` entry from the `PYTHONPATH` when starting Spyder.

### Issue(s) Resolved

See https://github.com/spyder-ide/spyder/issues/20658#issuecomment-1468338520

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: dalthviz

### CODE REVIEW

File: spyder/app/start.py [hunk 1]
```diff
@@ -12,10 +12,12 @@
 import sys
 if os.environ.get('PYTHONPATH'):
     for path in os.environ['PYTHONPATH'].split(os.pathsep):
-        try:
-            sys.path.remove(path.rstrip(os.sep))
-        except ValueError:
-            pass
+        if 'pkgs' not in path:
+            # Don't remove pynsist installer entry for 'pkgs' directory
```
Comment (ccordoba12) [inline spyder/app/start.py:16]:
```suggestion
        if os.name == 'nt' and 'pkgs' in path:
            # Don't remove pynsist installer entry for 'pkgs' directory
            continue
        else:
```

To be more specific about when we avoid removing a directory that contains `pkgs` in its path.

**Review by ccordoba12 — APPROVED** [review]
Left a small suggestion for you @dalthviz, otherwise looks good to me.

**Comment by ccordoba12** [comment]
@mrclary, it seems one of the patches for the Linux installer is not applying cleanly now. What can we can about that?

**Comment by mrclary** [comment]
It is probably because there is a merge conflict with the patch. This should be resolved by rebasing the patch branch and resolving any conflicts. I'll take a look to be sure and fix any issues I find.

**Comment by mrclary** [comment]
Regarding this PR, why does the Windows Spyder application require adding package paths to the user's PYTHONPATH? That seems unreliable as PYTHONPATH is intended for user customization and may be modified for any reason. Will the conda-based installer for Windows require use of PYTHONPATH as well?

**Comment by mrclary** [comment]
Okay, there is a direct merge conflict between the patch and the changes in this PR.
My recommendation is to move forward with this PR and ignore the failing conda-based installer failure. After this PR is merged, I'll rebase the patch branch and resolve the conflict, verifying that the conda-based Linux installer successfully builds.

**Comment by mrclary** [comment]
So `pynsist` creates the `Spyder-launch.pyw` script you listed?
It makes sense to me that `pkgdir` should be added to `sys.path`
```python
sys.path.insert(0, pkgdir)
# Ensure .pth files in pkgdir are handled properly
site.addsitedir(pkgdir)
```
But I don't know why it is added to `os.environ['PYTHONPATH']`
```python
os.environ['PYTHONPATH'] = pkgdir + os.pathsep + os.environ.get('PYTHONPATH', '')
```

Yes, the conda-based installers are exactly a conda installation, just in a non-standard location. You asked elsewhere whether we could use an nsist template for the conda-based installer, and I believe that we can. I don't know exactly what the template will do compared to the our existing Windows installer, e.g. will it generate the same `Spyder-launch.pyw`? I suppose we'll just have to cross that bridge when we get to it.

**Comment by dalthviz** [comment]
@ccordoba12 @mrclary should we merge this one as it is, right?

**Comment by mrclary** [comment]
> @ccordoba12 @mrclary should we merge this one as it is, right?

I think so. I'll resolve conflicts with the conda-based installer patch branch after this PR is merged.
I did not intend to hijack or delay this PR with my most recent questions/comments regarding PYTHONPATH.

**Comment by dalthviz** [comment]
> I think so. I'll resolve conflicts with the conda-based installer patch branch after this PR is merged.
:+1:
> I did not intend to hijack or delay this PR with my most recent questions/comments regarding PYTHONPATH.
Not at all! And in fact, quite interesting questions to think about in order to understand better what we need to do to migrate from the old installers :+1:

**Review by ccordoba12 — APPROVED** [review]
Thanks @dalthviz!

**Pull request merged**